src/wast-parser.cc

```diff
@@ -794,7 +801,12 @@ Result WastParser::ParseDataModuleField(Module* module) {
   auto field = MakeUnique<DataSegmentModuleField>(loc);
   EXPECT(Data);
   ParseVarOpt(&field->data_segment.memory_var, Var(0, loc));
```

Comment:
`passive` should be optionally parsed first, and if set the memory index should not be parsed.

See the spec [here](https://github.com/WebAssembly/bulk-memory-operations/blob/master/document/core/text/modules.rst#data-segments) (though it's kinda hard to read)